libazureinit/src/goalstate.rs

```diff
@@ -252,10 +252,85 @@ fn build_report_health_file(goalstate: Goalstate) -> String {
     )
 }
 
+#[instrument(err, skip_all)]
+pub async fn report_failure(
```

Comment:
from a high level, goalstate is a dependency for health (for the goalstate incarnation id), but otherwise unrelated.  i would suggest moving this logic into health.rs

so we would have something like:

health::report_ready()
health::report_failure()

maybe someday:
heath::report_progress()

report ready is already here so do what you will, just a thought.